<!DOCTYPE html>
<html lang="en-us">
<head>

  <meta charset="utf-8">
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="theme" content="hugo-academic">
  <meta name="generator" content="Hugo 0.42.2" />
  <meta name="author" content="Steven Xijin Ge">

  
  
  
  
    
      
    
  
  <meta name="description" content="Update. 11/13/17 Thanks to Mike Love for pointing out an error. This is just my understanding. There maybe other errors.
To allow complex study designs in iDEP(http://ge-lab.org/idep/), I tried to understand how factoral models are built and the desired results are extracted from DESeq2. The following is based on the help document from the resutls( ) function in DESeq2, plus some of Mike Love&rsquo;s answers to questions from users.">

  
  <link rel="alternate" hreflang="en-us" href="/post/deseq2-experimental-design-and-interpretation/">

  


  

  
  
  <meta name="theme-color" content="#0095eb">
  
  
  
  
    
  
  
    
    
      
        <link rel="stylesheet" href="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/styles/github.min.css">
      
    
  
  
  <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha512-6MXa8B6uaO18Hid6blRMetEIoPqHf7Ux1tnyIQdpt9qI5OACx7C+O3IVTr98vwGnlcg0LOLa02i9Y1HpVhlfiw==" crossorigin="anonymous">
  <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/academicons/1.8.6/css/academicons.min.css" integrity="sha256-uFVgMKfistnJAfoCUQigIl+JfUaP47GrRKjf6CTPVmw=" crossorigin="anonymous">
  <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css" integrity="sha512-SfTiTlX6kk+qitfevl/7LibUOeJWlt9rbyDn92a1DqWOw9vWG2MFoays0sgObmWazO5BQPiFucnnEAjpAB+/Sw==" crossorigin="anonymous">
  <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/fancybox/3.2.5/jquery.fancybox.min.css" integrity="sha256-ygkqlh3CYSUri3LhQxzdcm0n1EQvH2Y+U5S2idbLtxs=" crossorigin="anonymous">
  
  
  
  
  <link rel="stylesheet" href="//fonts.googleapis.com/css?family=Montserrat:400,700%7cRoboto:400,400italic,700%7cRoboto&#43;Mono">
  
  <link rel="stylesheet" href="/styles.css">
  

  
  
    <script>
      window.ga=window.ga||function(){(ga.q=ga.q||[]).push(arguments)};ga.l=+new Date;
      ga('create', 'UA-87863704-4', 'auto');
      
      ga('require', 'eventTracker');
      ga('require', 'outboundLinkTracker');
      ga('require', 'urlChangeTracker');
      ga('send', 'pageview');
    </script>
    <script async src="//www.google-analytics.com/analytics.js"></script>
    
    <script async src="https://cdnjs.cloudflare.com/ajax/libs/autotrack/2.4.1/autotrack.js" integrity="sha512-HUmooslVKj4m6OBu0OgzjXXr+QuFYy/k7eLI5jdeEy/F4RSgMn6XRWRGkFi5IFaFgy7uFTkegp3Z0XnJf3Jq+g==" crossorigin="anonymous"></script>
    
  
  

  
  <link rel="alternate" href="/index.xml" type="application/rss+xml" title="In Data We Trust">
  <link rel="feed" href="/index.xml" type="application/rss+xml" title="In Data We Trust">
  

  <link rel="manifest" href="/site.webmanifest">
  <link rel="icon" type="image/png" href="/img/icon.png">
  <link rel="apple-touch-icon" type="image/png" href="/img/icon-192.png">

  <link rel="canonical" href="/post/deseq2-experimental-design-and-interpretation/">

  <meta property="twitter:card" content="summary_large_image">
  
  <meta property="twitter:site" content="@StevenXGe">
  <meta property="twitter:creator" content="@StevenXGe">
  
  <meta property="og:site_name" content="In Data We Trust">
  <meta property="og:url" content="/post/deseq2-experimental-design-and-interpretation/">
  <meta property="og:title" content="DESeq2 experimental design and interpretation | In Data We Trust">
  <meta property="og:description" content="Update. 11/13/17 Thanks to Mike Love for pointing out an error. This is just my understanding. There maybe other errors.
To allow complex study designs in iDEP(http://ge-lab.org/idep/), I tried to understand how factoral models are built and the desired results are extracted from DESeq2. The following is based on the help document from the resutls( ) function in DESeq2, plus some of Mike Love&rsquo;s answers to questions from users.">
  <meta property="og:locale" content="en-us">
  
  <meta property="article:published_time" content="2018-07-02T00:00:00&#43;00:00">
  
  <meta property="article:modified_time" content="2018-07-02T00:00:00&#43;00:00">
  

  

  

  <title>DESeq2 experimental design and interpretation | In Data We Trust</title>

</head>
<body id="top" data-spy="scroll" data-target="#toc" data-offset="71" >

<nav class="navbar navbar-default navbar-fixed-top" id="navbar-main">
  <div class="container">

    
    <div class="navbar-header">
      
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse"
              data-target=".navbar-collapse" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      
      <a class="navbar-brand" href="/">In Data We Trust</a>
    </div>

    
    <div class="collapse navbar-collapse">

      
      
      <ul class="nav navbar-nav navbar-right">
        

        
        
        
        
        
          
        

        <li class="nav-item">
          <a href="/#posts">
            
            <span>Posts</span>
            
          </a>
        </li>

        
        

        
        
        
        
        
          
        

        <li class="nav-item">
          <a href="/#contact">
            
            <span>About</span>
            
          </a>
        </li>

        
        
      

      
      </ul>

    </div>
  </div>
</nav>


<article class="article" itemscope itemtype="http://schema.org/Article">

  


  <div class="article-container">
    <h1 itemprop="name">DESeq2 experimental design and interpretation</h1>

    

<div class="article-metadata">

  

  <span class="article-date">
    
    <meta content="2018-07-02 00:00:00 &#43;0000 UTC" itemprop="datePublished">
    <time datetime="2018-07-02 00:00:00 &#43;0000 UTC" itemprop="dateModified">
      Jul 2, 2018
    </time>
  </span>
  <span itemscope itemprop="author publisher" itemtype="http://schema.org/Person">
    <meta itemprop="name" content="Steven Xijin Ge">
  </span>

  
  <span class="middot-divider"></span>
  <span class="article-reading-time">
    13 min read
  </span>
  

  
  
  <span class="middot-divider"></span>
  <a href="/post/deseq2-experimental-design-and-interpretation/#disqus_thread"></a>
  

  
  
  
  <span class="middot-divider"></span>
  <span class="article-categories">
    <i class="fa fa-folder"></i>
    
    <a href="/categories/r/">R</a
    >
    
  </span>
  
  

  
  
<div class="share-box" aria-hidden="true">
  <ul class="share">
    <li>
      <a class="twitter"
         href="https://twitter.com/intent/tweet?text=DESeq2%20experimental%20design%20and%20interpretation&amp;url=%2fpost%2fdeseq2-experimental-design-and-interpretation%2f"
         target="_blank" rel="noopener">
        <i class="fa fa-twitter"></i>
      </a>
    </li>
    <li>
      <a class="facebook"
         href="https://www.facebook.com/sharer.php?u=%2fpost%2fdeseq2-experimental-design-and-interpretation%2f"
         target="_blank" rel="noopener">
        <i class="fa fa-facebook"></i>
      </a>
    </li>
    <li>
      <a class="linkedin"
         href="https://www.linkedin.com/shareArticle?mini=true&amp;url=%2fpost%2fdeseq2-experimental-design-and-interpretation%2f&amp;title=DESeq2%20experimental%20design%20and%20interpretation"
         target="_blank" rel="noopener">
        <i class="fa fa-linkedin"></i>
      </a>
    </li>
    <li>
      <a class="weibo"
         href="http://service.weibo.com/share/share.php?url=%2fpost%2fdeseq2-experimental-design-and-interpretation%2f&amp;title=DESeq2%20experimental%20design%20and%20interpretation"
         target="_blank" rel="noopener">
        <i class="fa fa-weibo"></i>
      </a>
    </li>
    <li>
      <a class="email"
         href="mailto:?subject=DESeq2%20experimental%20design%20and%20interpretation&amp;body=%2fpost%2fdeseq2-experimental-design-and-interpretation%2f">
        <i class="fa fa-envelope"></i>
      </a>
    </li>
  </ul>
</div>


  

</div>


    <div class="article-style" itemprop="articleBody">
      

<p><img src="/post/2018-07-02-deseq2-experimental-design-and-interpretation_files/factors5.jpg" alt="" /></p>

<p><em>Update. 11/13/17  Thanks to Mike Love for pointing out an error. This is just my understanding. There maybe other errors.</em></p>

<p>To allow complex study designs in iDEP(<a href="http://ge-lab.org/idep/" target="_blank">http://ge-lab.org/idep/</a>), I tried to understand how factoral models are built and the desired results are extracted from DESeq2.  The following is based on the help document from the resutls( ) function in DESeq2, plus some of Mike Love&rsquo;s answers to questions from users.</p>

<p>An important point I want to make is the interpretation of results is tricky when the study design involve multiple factors.  See figure above for an example involving 3 genotypes under 3  conditions.  Similar to regression analysis in R, the <em>reference levels</em> for categorical factors forms the foundation of our intereptation. Yet, by default, they are determined alphabetically.</p>

<p>Before runing DESeq2, it is essential to choose appropriate reference levels for each factors. This can be done by the relevel( ) function in R. Reference level is the baseline level of a factor that forms the basis of meaningful comparisons. In a wildtype vs. mutant experiment, &ldquo;wild-type&rdquo; is the reference level. In treated vs. untreated, the reference level is obviously untreated. More details in Exmple 3.</p>

<p>iDEP provides a GUI to DESeq2 for most experimental desings. It also provides convienent interface for exploratory data anlysis and pathway analysis. Try it at <a href="http://ge-lab.org/idep/" target="_blank">http://ge-lab.org/idep/</a></p>

<h1 id="example-1-two-group-comparison">Example 1: two-group comparison</h1>

<p>First make some example data.</p>

<pre><code class="language-r">library(DESeq2)
dds &lt;- makeExampleDESeqDataSet(n=10000,m=6)
assay(dds)[ 1:10,]
</code></pre>

<pre><code>##        sample1 sample2 sample3 sample4 sample5 sample6
## gene1        6       4      11       1       2      13
## gene2        9      12      23      13      14      28
## gene3       58     121     173     178     118      97
## gene4        0       4       0       3       8       3
## gene5       27       3       6       9       8      12
## gene6       48       8      35      38      21      13
## gene7       36      50      61      52      44      22
## gene8        6       8      16      14      18      19
## gene9      214     266     419     198     157     166
## gene10      20      12      16      12      16       2
</code></pre>

<p>This is a very simple experiment design with two conditions.</p>

<pre><code class="language-r">colData(dds)
</code></pre>

<pre><code>## DataFrame with 6 rows and 1 column
##         condition
##          &lt;factor&gt;
## sample1         A
## sample2         A
## sample3         A
## sample4         B
## sample5         B
## sample6         B
</code></pre>

<pre><code class="language-r">dds &lt;- DESeq(dds)
resultsNames(dds)
</code></pre>

<pre><code>## [1] &quot;Intercept&quot;        &quot;condition_B_vs_A&quot;
</code></pre>

<p>This shows the results available. Note that by default, R will choose a reference level for factors based on alphabetical order. Here A is the referece level. Fold change is defined as B compaired with A. To change reference levels, try the relevel( ) function.</p>

<pre><code class="language-r">res &lt;- results(dds, contrast=c(&quot;condition&quot;,&quot;B&quot;,&quot;A&quot;))
res &lt;- res[order(res$padj),]
library(knitr)
kable(res[1:5,-(3:4)])
</code></pre>

<table>
<thead>
<tr>
<th align="left"></th>
<th align="right">baseMean</th>
<th align="right">log2FoldChange</th>
<th align="right">pvalue</th>
<th align="right">padj</th>
</tr>
</thead>

<tbody>
<tr>
<td align="left">gene9056</td>
<td align="right">360.168909</td>
<td align="right">-2.045379</td>
<td align="right">0.0000000</td>
<td align="right">0.0001372</td>
</tr>

<tr>
<td align="left">gene3087</td>
<td align="right">43.897516</td>
<td align="right">-2.203303</td>
<td align="right">0.0000173</td>
<td align="right">0.0857004</td>
</tr>

<tr>
<td align="left">gene3763</td>
<td align="right">72.409877</td>
<td align="right">-1.834787</td>
<td align="right">0.0000433</td>
<td align="right">0.1434364</td>
</tr>

<tr>
<td align="left">gene2054</td>
<td align="right">322.494963</td>
<td align="right">1.537408</td>
<td align="right">0.0000682</td>
<td align="right">0.1692730</td>
</tr>

<tr>
<td align="left">gene4617</td>
<td align="right">6.227415</td>
<td align="right">6.125238</td>
<td align="right">0.0002015</td>
<td align="right">0.4001027</td>
</tr>
</tbody>
</table>

<p>If we want to use B as control and define fold change using B as baseline. Then we can do this:</p>

<pre><code class="language-r">res &lt;- results(dds, contrast=c(&quot;condition&quot;,&quot;A&quot;,&quot;B&quot;))
ix = which.min(res$padj)
res &lt;- res[order(res$padj),]
kable(res[1:5,-(3:4)])
</code></pre>

<table>
<thead>
<tr>
<th align="left"></th>
<th align="right">baseMean</th>
<th align="right">log2FoldChange</th>
<th align="right">pvalue</th>
<th align="right">padj</th>
</tr>
</thead>

<tbody>
<tr>
<td align="left">gene9056</td>
<td align="right">360.168909</td>
<td align="right">2.045379</td>
<td align="right">0.0000000</td>
<td align="right">0.0001372</td>
</tr>

<tr>
<td align="left">gene3087</td>
<td align="right">43.897516</td>
<td align="right">2.203303</td>
<td align="right">0.0000173</td>
<td align="right">0.0857004</td>
</tr>

<tr>
<td align="left">gene3763</td>
<td align="right">72.409877</td>
<td align="right">1.834787</td>
<td align="right">0.0000433</td>
<td align="right">0.1434364</td>
</tr>

<tr>
<td align="left">gene2054</td>
<td align="right">322.494963</td>
<td align="right">-1.537408</td>
<td align="right">0.0000682</td>
<td align="right">0.1692730</td>
</tr>

<tr>
<td align="left">gene4617</td>
<td align="right">6.227415</td>
<td align="right">-6.125238</td>
<td align="right">0.0002015</td>
<td align="right">0.4001027</td>
</tr>
</tbody>
</table>

<p>As you can see, the fold-change are completely opposite direction. Here we show the most significant gene.</p>

<pre><code class="language-r">barplot(assay(dds)[ix,],las=2, main=rownames(dds)[ ix  ]  )
</code></pre>

<p><img src="/post/2018-07-02-deseq2-experimental-design-and-interpretation_files/figure-html/unnamed-chunk-6-1.png" width="576" /></p>

<p>#Example 2: Multiple groups
Suppose we have three groups A, B, and C.</p>

<pre><code class="language-r">dds &lt;- makeExampleDESeqDataSet(n=100,m=6)
dds$condition &lt;- factor( c( &quot;A&quot;,&quot;A&quot;,&quot;B&quot;,&quot;B&quot;,&quot;C&quot;,&quot;C&quot;) )
dds &lt;- DESeq(dds)
res = results(dds, contrast=c(&quot;condition&quot;,&quot;C&quot;,&quot;A&quot;))
res &lt;- res[order(res$padj),]
kable(res[1:5,-(3:4)])
</code></pre>

<table>
<thead>
<tr>
<th align="left"></th>
<th align="right">baseMean</th>
<th align="right">log2FoldChange</th>
<th align="right">pvalue</th>
<th align="right">padj</th>
</tr>
</thead>

<tbody>
<tr>
<td align="left">gene2</td>
<td align="right">3.634986</td>
<td align="right">-5.101773</td>
<td align="right">0.0348679</td>
<td align="right">0.5515088</td>
</tr>

<tr>
<td align="left">gene20</td>
<td align="right">4.678176</td>
<td align="right">-4.490982</td>
<td align="right">0.0445664</td>
<td align="right">0.5515088</td>
</tr>

<tr>
<td align="left">gene34</td>
<td align="right">56.068672</td>
<td align="right">-1.462155</td>
<td align="right">0.0167820</td>
<td align="right">0.5515088</td>
</tr>

<tr>
<td align="left">gene35</td>
<td align="right">537.847175</td>
<td align="right">-1.177240</td>
<td align="right">0.0087913</td>
<td align="right">0.5515088</td>
</tr>

<tr>
<td align="left">gene41</td>
<td align="right">93.967810</td>
<td align="right">1.064734</td>
<td align="right">0.0412034</td>
<td align="right">0.5515088</td>
</tr>
</tbody>
</table>

<h1 id="example-3-two-conditions-two-genotypes-with-an-interaction-term">Example 3: two conditions, two genotypes, with an interaction term</h1>

<p>Here we have two genotypes, wild-type (WT), and mutant (MU). Two conditions, control (Ctrl) and treated (Trt). We are interested in the responses of both wild-type and mutant to treatment. We are also interested in the differences in response between genotypes, which is captured by the interaction term in linear models.</p>

<p>First, we construct example data. Note that we changed sample names from &ldquo;sample1&rdquo; to &ldquo;Wt_Ctrl_1&rdquo;, according to the two factors.</p>

<pre><code class="language-r">dds &lt;- makeExampleDESeqDataSet(n=10000,m=12)
dds$condition &lt;- factor( c( rep(&quot;Ctrl&quot;,6), rep(&quot;Trt&quot;,6) ) )
dds$genotype &lt;- factor(rep(rep(c(&quot;WT&quot;,&quot;MU&quot;),each=3),2))
colnames(dds) &lt;- paste(as.character( dds$genotype),as.character( dds$condition),rownames(colData(dds)),  sep=&quot;_&quot;  )
colnames(dds) = gsub(&quot;sample&quot;,&quot;&quot;,colnames(dds))
kable(assay(dds)[1:5,])
</code></pre>

<table>
<thead>
<tr>
<th align="left"></th>
<th align="right">WT_Ctrl_1</th>
<th align="right">WT_Ctrl_2</th>
<th align="right">WT_Ctrl_3</th>
<th align="right">MU_Ctrl_4</th>
<th align="right">MU_Ctrl_5</th>
<th align="right">MU_Ctrl_6</th>
<th align="right">WT_Trt_7</th>
<th align="right">WT_Trt_8</th>
<th align="right">WT_Trt_9</th>
<th align="right">MU_Trt_10</th>
<th align="right">MU_Trt_11</th>
<th align="right">MU_Trt_12</th>
</tr>
</thead>

<tbody>
<tr>
<td align="left">gene1</td>
<td align="right">81</td>
<td align="right">47</td>
<td align="right">135</td>
<td align="right">72</td>
<td align="right">81</td>
<td align="right">80</td>
<td align="right">76</td>
<td align="right">57</td>
<td align="right">52</td>
<td align="right">49</td>
<td align="right">64</td>
<td align="right">57</td>
</tr>

<tr>
<td align="left">gene2</td>
<td align="right">70</td>
<td align="right">77</td>
<td align="right">94</td>
<td align="right">54</td>
<td align="right">54</td>
<td align="right">36</td>
<td align="right">51</td>
<td align="right">66</td>
<td align="right">57</td>
<td align="right">51</td>
<td align="right">47</td>
<td align="right">67</td>
</tr>

<tr>
<td align="left">gene3</td>
<td align="right">16</td>
<td align="right">37</td>
<td align="right">8</td>
<td align="right">28</td>
<td align="right">4</td>
<td align="right">36</td>
<td align="right">14</td>
<td align="right">10</td>
<td align="right">7</td>
<td align="right">11</td>
<td align="right">40</td>
<td align="right">15</td>
</tr>

<tr>
<td align="left">gene4</td>
<td align="right">2</td>
<td align="right">5</td>
<td align="right">1</td>
<td align="right">3</td>
<td align="right">9</td>
<td align="right">2</td>
<td align="right">1</td>
<td align="right">5</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">14</td>
<td align="right">8</td>
</tr>

<tr>
<td align="left">gene5</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">5</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
</tbody>
</table>

<pre><code class="language-r">kable( colData(dds))
</code></pre>

<table>
<thead>
<tr>
<th align="left"></th>
<th align="left">condition</th>
<th align="left">genotype</th>
</tr>
</thead>

<tbody>
<tr>
<td align="left">WT_Ctrl_1</td>
<td align="left">Ctrl</td>
<td align="left">WT</td>
</tr>

<tr>
<td align="left">WT_Ctrl_2</td>
<td align="left">Ctrl</td>
<td align="left">WT</td>
</tr>

<tr>
<td align="left">WT_Ctrl_3</td>
<td align="left">Ctrl</td>
<td align="left">WT</td>
</tr>

<tr>
<td align="left">MU_Ctrl_4</td>
<td align="left">Ctrl</td>
<td align="left">MU</td>
</tr>

<tr>
<td align="left">MU_Ctrl_5</td>
<td align="left">Ctrl</td>
<td align="left">MU</td>
</tr>

<tr>
<td align="left">MU_Ctrl_6</td>
<td align="left">Ctrl</td>
<td align="left">MU</td>
</tr>

<tr>
<td align="left">WT_Trt_7</td>
<td align="left">Trt</td>
<td align="left">WT</td>
</tr>

<tr>
<td align="left">WT_Trt_8</td>
<td align="left">Trt</td>
<td align="left">WT</td>
</tr>

<tr>
<td align="left">WT_Trt_9</td>
<td align="left">Trt</td>
<td align="left">WT</td>
</tr>

<tr>
<td align="left">MU_Trt_10</td>
<td align="left">Trt</td>
<td align="left">MU</td>
</tr>

<tr>
<td align="left">MU_Trt_11</td>
<td align="left">Trt</td>
<td align="left">MU</td>
</tr>

<tr>
<td align="left">MU_Trt_12</td>
<td align="left">Trt</td>
<td align="left">MU</td>
</tr>
</tbody>
</table>

<p>Check reference levels:</p>

<pre><code class="language-r">dds$condition
</code></pre>

<pre><code>##  [1] Ctrl Ctrl Ctrl Ctrl Ctrl Ctrl Trt  Trt  Trt  Trt  Trt  Trt 
## Levels: Ctrl Trt
</code></pre>

<p>As you could see, &ldquo;Ctrl&rdquo; apeared first in the 2nd line, indicating it is the reference level for factor condition, as we can expect based on alphabetical order. This is what we want and we do not need to do anything.</p>

<pre><code class="language-r">dds$genotype
</code></pre>

<pre><code>##  [1] WT WT WT MU MU MU WT WT WT MU MU MU
## Levels: MU WT
</code></pre>

<p>But  &ldquo;Mu&rdquo; is the reference level for genotype, which is will give us results difficult to interpret. We need to change it.</p>

<pre><code class="language-r">dds$genotype = relevel( dds$genotype, &quot;WT&quot;)
dds$genotype
</code></pre>

<pre><code>##  [1] WT WT WT MU MU MU WT WT WT MU MU MU
## Levels: WT MU
</code></pre>

<p>Set up the model, and run DESeq2:</p>

<pre><code class="language-r">design(dds) &lt;- ~ genotype + condition + genotype:condition
dds &lt;- DESeq(dds) 
resultsNames(dds)
</code></pre>

<pre><code>## [1] &quot;Intercept&quot;               &quot;genotype_MU_vs_WT&quot;      
## [3] &quot;condition_Trt_vs_Ctrl&quot;   &quot;genotypeMU.conditionTrt&quot;
</code></pre>

<p>Below, we are going to use the combination of the different results (&ldquo;genotype_MU_vs_WT&rdquo;, &ldquo;condition_Trt_vs_Ctrl&rdquo;, &ldquo;genotypeMU.conditionTrt&rdquo; ) to derive biologically meaningful comparisons.</p>

<p>##The  effect of treatment in wild-type (the main effect).</p>

<pre><code class="language-r">res = results(dds, contrast=c(&quot;condition&quot;,&quot;Trt&quot;,&quot;Ctrl&quot;))
ix = which.min(res$padj) # most significant
res &lt;- res[order(res$padj),] # sort
kable(res[1:5,-(3:4)])
</code></pre>

<table>
<thead>
<tr>
<th align="left"></th>
<th align="right">baseMean</th>
<th align="right">log2FoldChange</th>
<th align="right">pvalue</th>
<th align="right">padj</th>
</tr>
</thead>

<tbody>
<tr>
<td align="left">gene275</td>
<td align="right">25.38836</td>
<td align="right">2.607896</td>
<td align="right">0.0000588</td>
<td align="right">0.2684841</td>
</tr>

<tr>
<td align="left">gene2744</td>
<td align="right">101.02954</td>
<td align="right">-1.670837</td>
<td align="right">0.0001099</td>
<td align="right">0.2684841</td>
</tr>

<tr>
<td align="left">gene5441</td>
<td align="right">58.67469</td>
<td align="right">1.758921</td>
<td align="right">0.0001261</td>
<td align="right">0.2684841</td>
</tr>

<tr>
<td align="left">gene7021</td>
<td align="right">74.29359</td>
<td align="right">1.610853</td>
<td align="right">0.0001345</td>
<td align="right">0.2684841</td>
</tr>

<tr>
<td align="left">gene7795</td>
<td align="right">326.43308</td>
<td align="right">-1.624323</td>
<td align="right">0.0000407</td>
<td align="right">0.2684841</td>
</tr>
</tbody>
</table>

<p>This is for WT, treated compared with untreated. Note that WT is not mentioned, because it is the reference level. In other words, this is the difference between samples No. 7-9, compared with samples No. 1-3.</p>

<p>Here we show the most significant gene.</p>

<pre><code class="language-r">barplot(assay(dds)[ix,],las=2, main=rownames(dds)[ ix  ]  )
</code></pre>

<p><img src="/post/2018-07-02-deseq2-experimental-design-and-interpretation_files/figure-html/unnamed-chunk-14-1.png" width="576" /></p>

<p>##The effect of treatment in mutant
This is, by definition, the main effect <em>plus</em> the interaction term (the extra condition effect in genotype Mutant compared to genotype WT).</p>

<pre><code class="language-r">res &lt;- results(dds, list( c(&quot;condition_Trt_vs_Ctrl&quot;,&quot;genotypeMU.conditionTrt&quot;) ))
ix = which.min(res$padj) # most significant
res &lt;- res[order(res$padj),] # sort
kable(res[1:5,-(3:4)])
</code></pre>

<table>
<thead>
<tr>
<th align="left"></th>
<th align="right">baseMean</th>
<th align="right">log2FoldChange</th>
<th align="right">pvalue</th>
<th align="right">padj</th>
</tr>
</thead>

<tbody>
<tr>
<td align="left">gene5102</td>
<td align="right">18.69017</td>
<td align="right">4.757057</td>
<td align="right">1.60e-06</td>
<td align="right">0.0156912</td>
</tr>

<tr>
<td align="left">gene7367</td>
<td align="right">170.69259</td>
<td align="right">1.481016</td>
<td align="right">1.19e-05</td>
<td align="right">0.0396827</td>
</tr>

<tr>
<td align="left">gene8351</td>
<td align="right">27.77227</td>
<td align="right">3.033622</td>
<td align="right">9.80e-06</td>
<td align="right">0.0396827</td>
</tr>

<tr>
<td align="left">gene8034</td>
<td align="right">53.34342</td>
<td align="right">-1.841023</td>
<td align="right">1.62e-05</td>
<td align="right">0.0403724</td>
</tr>

<tr>
<td align="left">gene7272</td>
<td align="right">92.82351</td>
<td align="right">-1.414967</td>
<td align="right">6.70e-05</td>
<td align="right">0.1125817</td>
</tr>
</tbody>
</table>

<p>Note that it has to be <em>list</em>( c(&ldquo;condition_Trt_vs_Ctrl&rdquo;,&ldquo;genotypeMU.conditionTrt&rdquo;) ). If list( ) is ommitted, the results would be drastically different. Perhaps only the first coefficient is used.</p>

<p>This measures the effect of treatment in mutant. In other words, samples No. 10-12 compared with samples No. 4-6.</p>

<p>Here we show the most significant gene, which is downregulated expressed in samples 10-12, than samples 4-6 , as expected.</p>

<pre><code class="language-r">barplot(assay(dds)[ix,],las=2, main=rownames(dds)[ ix  ]  )
</code></pre>

<p><img src="/post/2018-07-02-deseq2-experimental-design-and-interpretation_files/figure-html/unnamed-chunk-16-1.png" width="576" /></p>

<p>##What is the difference between mutant and wild-type without treatment?</p>

<p>As Ctrl is the reference level, we can just retrieve the &ldquo;genotype_MU_vs_WT&rdquo;.</p>

<pre><code class="language-r">res = results(dds, contrast=c(&quot;genotype&quot;,&quot;MU&quot;,&quot;WT&quot;))
ix = which.min(res$padj) # most significant
res &lt;- res[order(res$padj),] # sort
kable(res[1:5,-(3:4)])
</code></pre>

<table>
<thead>
<tr>
<th align="left"></th>
<th align="right">baseMean</th>
<th align="right">log2FoldChange</th>
<th align="right">pvalue</th>
<th align="right">padj</th>
</tr>
</thead>

<tbody>
<tr>
<td align="left">gene5102</td>
<td align="right">18.69017</td>
<td align="right">-4.714519</td>
<td align="right">0.0000020</td>
<td align="right">0.0195596</td>
</tr>

<tr>
<td align="left">gene2289</td>
<td align="right">45.04711</td>
<td align="right">1.763615</td>
<td align="right">0.0000333</td>
<td align="right">0.1218107</td>
</tr>

<tr>
<td align="left">gene3388</td>
<td align="right">122.85582</td>
<td align="right">-1.529323</td>
<td align="right">0.0000366</td>
<td align="right">0.1218107</td>
</tr>

<tr>
<td align="left">gene915</td>
<td align="right">39.03250</td>
<td align="right">2.104003</td>
<td align="right">0.0001133</td>
<td align="right">0.2374861</td>
</tr>

<tr>
<td align="left">gene8351</td>
<td align="right">27.77227</td>
<td align="right">-2.653182</td>
<td align="right">0.0001190</td>
<td align="right">0.2374861</td>
</tr>
</tbody>
</table>

<p>In other words, this is the samples No.4-6  compared with No. 1-3. Here we show the most significant gene.</p>

<pre><code class="language-r">barplot(assay(dds)[ix,],las=2, main=rownames(dds)[ ix  ]  )
</code></pre>

<p><img src="/post/2018-07-02-deseq2-experimental-design-and-interpretation_files/figure-html/unnamed-chunk-18-1.png" width="576" /></p>

<p>##With treatment, what is the difference between mutant and wild-type?</p>

<pre><code class="language-r">res = results(dds, list( c(&quot;genotype_MU_vs_WT&quot;,&quot;genotypeMU.conditionTrt&quot;) ))
ix = which.min(res$padj) # most significant
res &lt;- res[order(res$padj),] # sort
kable(res[1:5,-(3:4)])
</code></pre>

<table>
<thead>
<tr>
<th align="left"></th>
<th align="right">baseMean</th>
<th align="right">log2FoldChange</th>
<th align="right">pvalue</th>
<th align="right">padj</th>
</tr>
</thead>

<tbody>
<tr>
<td align="left">gene4127</td>
<td align="right">50.03677</td>
<td align="right">1.925094</td>
<td align="right">0.0000091</td>
<td align="right">0.0910879</td>
</tr>

<tr>
<td align="left">gene3879</td>
<td align="right">34.67704</td>
<td align="right">-2.133203</td>
<td align="right">0.0000283</td>
<td align="right">0.0940393</td>
</tr>

<tr>
<td align="left">gene4799</td>
<td align="right">24.11059</td>
<td align="right">-2.459345</td>
<td align="right">0.0000197</td>
<td align="right">0.0940393</td>
</tr>

<tr>
<td align="left">gene5441</td>
<td align="right">58.67469</td>
<td align="right">-1.744769</td>
<td align="right">0.0001418</td>
<td align="right">0.2357639</td>
</tr>

<tr>
<td align="left">gene5926</td>
<td align="right">161.16737</td>
<td align="right">1.524163</td>
<td align="right">0.0001339</td>
<td align="right">0.2357639</td>
</tr>
</tbody>
</table>

<p>This gives us the difference between genotype MU and WT, under condition Trt. In other words, this is the sampless No. 10-12 compared with samples 7-9.</p>

<p>Here we show the most significant gene.</p>

<pre><code class="language-r">barplot(assay(dds)[ix,],las=2, main=rownames(dds)[ ix  ]  )
</code></pre>

<p><img src="/post/2018-07-02-deseq2-experimental-design-and-interpretation_files/figure-html/unnamed-chunk-20-1.png" width="576" /></p>

<h2 id="the-different-response-in-genotypes-interaction-term">The different response in genotypes (interaction term)</h2>

<p>Is the  effect of treatment <em>different</em> across genotypes? This is the interaction term.</p>

<pre><code class="language-r">res = results(dds, name=&quot;genotypeMU.conditionTrt&quot;)
ix = which.min(res$padj) # most significant
res &lt;- res[order(res$padj),] # sort
kable(res[1:5,-(3:4)])
</code></pre>

<table>
<thead>
<tr>
<th align="left"></th>
<th align="right">baseMean</th>
<th align="right">log2FoldChange</th>
<th align="right">pvalue</th>
<th align="right">padj</th>
</tr>
</thead>

<tbody>
<tr>
<td align="left">gene5102</td>
<td align="right">18.69017</td>
<td align="right">7.395771</td>
<td align="right">0.0000000</td>
<td align="right">0.0000341</td>
</tr>

<tr>
<td align="left">gene5441</td>
<td align="right">58.67469</td>
<td align="right">-3.297673</td>
<td align="right">0.0000004</td>
<td align="right">0.0019104</td>
</tr>

<tr>
<td align="left">gene8034</td>
<td align="right">53.34342</td>
<td align="right">-2.564030</td>
<td align="right">0.0000205</td>
<td align="right">0.0682558</td>
</tr>

<tr>
<td align="left">gene8858</td>
<td align="right">14.68650</td>
<td align="right">4.703936</td>
<td align="right">0.0000297</td>
<td align="right">0.0742076</td>
</tr>

<tr>
<td align="left">gene4799</td>
<td align="right">24.11059</td>
<td align="right">-2.990101</td>
<td align="right">0.0001234</td>
<td align="right">0.2463661</td>
</tr>
</tbody>
</table>

<p>Here we show the most significant gene.</p>

<pre><code class="language-r">barplot(assay(dds)[ix,],las=2, main=rownames(dds)[ ix  ]  )
</code></pre>

<p><img src="/post/2018-07-02-deseq2-experimental-design-and-interpretation_files/figure-html/unnamed-chunk-22-1.png" width="576" /></p>

<p>In wild-type, this gene is upregulated by treatment, while for the  mutant, it is downregulated.</p>

<h1 id="example-4-two-conditionss-three-genotpes-with-interaction-terms">Example 4: Two conditionss, three genotpes with interaction terms</h1>

<pre><code class="language-r">dds &lt;- makeExampleDESeqDataSet(n=10000,m=18)
dds$genotype &lt;- factor(rep(rep(c(&quot;I&quot;,&quot;II&quot;,&quot;III&quot;),each=3),2))
colnames(dds) &lt;- paste(rownames(colData(dds)), as.character( dds$condition), as.character( dds$genotype),sep=&quot;_&quot;  )
colnames(dds) = gsub(&quot;sample&quot;,&quot;S&quot;,colnames(dds))
design(dds) &lt;- ~ genotype + condition + genotype:condition
dds &lt;- DESeq(dds)
kable( colData(dds))
</code></pre>

<table>
<thead>
<tr>
<th align="left"></th>
<th align="left">condition</th>
<th align="left">genotype</th>
<th align="right">sizeFactor</th>
</tr>
</thead>

<tbody>
<tr>
<td align="left">S1_A_I</td>
<td align="left">A</td>
<td align="left">I</td>
<td align="right">1.053964</td>
</tr>

<tr>
<td align="left">S2_A_I</td>
<td align="left">A</td>
<td align="left">I</td>
<td align="right">1.058935</td>
</tr>

<tr>
<td align="left">S3_A_I</td>
<td align="left">A</td>
<td align="left">I</td>
<td align="right">1.061629</td>
</tr>

<tr>
<td align="left">S4_A_II</td>
<td align="left">A</td>
<td align="left">II</td>
<td align="right">1.047347</td>
</tr>

<tr>
<td align="left">S5_A_II</td>
<td align="left">A</td>
<td align="left">II</td>
<td align="right">1.059813</td>
</tr>

<tr>
<td align="left">S6_A_II</td>
<td align="left">A</td>
<td align="left">II</td>
<td align="right">1.049639</td>
</tr>

<tr>
<td align="left">S7_A_III</td>
<td align="left">A</td>
<td align="left">III</td>
<td align="right">1.060271</td>
</tr>

<tr>
<td align="left">S8_A_III</td>
<td align="left">A</td>
<td align="left">III</td>
<td align="right">1.057331</td>
</tr>

<tr>
<td align="left">S9_A_III</td>
<td align="left">A</td>
<td align="left">III</td>
<td align="right">1.052053</td>
</tr>

<tr>
<td align="left">S10_B_I</td>
<td align="left">B</td>
<td align="left">I</td>
<td align="right">1.053989</td>
</tr>

<tr>
<td align="left">S11_B_I</td>
<td align="left">B</td>
<td align="left">I</td>
<td align="right">1.044408</td>
</tr>

<tr>
<td align="left">S12_B_I</td>
<td align="left">B</td>
<td align="left">I</td>
<td align="right">1.042376</td>
</tr>

<tr>
<td align="left">S13_B_II</td>
<td align="left">B</td>
<td align="left">II</td>
<td align="right">1.042354</td>
</tr>

<tr>
<td align="left">S14_B_II</td>
<td align="left">B</td>
<td align="left">II</td>
<td align="right">1.050094</td>
</tr>

<tr>
<td align="left">S15_B_II</td>
<td align="left">B</td>
<td align="left">II</td>
<td align="right">1.045868</td>
</tr>

<tr>
<td align="left">S16_B_III</td>
<td align="left">B</td>
<td align="left">III</td>
<td align="right">1.050772</td>
</tr>

<tr>
<td align="left">S17_B_III</td>
<td align="left">B</td>
<td align="left">III</td>
<td align="right">1.047934</td>
</tr>

<tr>
<td align="left">S18_B_III</td>
<td align="left">B</td>
<td align="left">III</td>
<td align="right">1.036983</td>
</tr>
</tbody>
</table>

<pre><code class="language-r">resultsNames(dds)
</code></pre>

<pre><code>## [1] &quot;Intercept&quot;              &quot;genotype_II_vs_I&quot;      
## [3] &quot;genotype_III_vs_I&quot;      &quot;condition_B_vs_A&quot;      
## [5] &quot;genotypeII.conditionB&quot;  &quot;genotypeIII.conditionB&quot;
</code></pre>

<h2 id="the-condition-effect-for-genotype-i-the-main-effect">The condition effect for genotype I (the main effect)</h2>

<pre><code class="language-r">res = results(dds, contrast=c(&quot;condition&quot;,&quot;B&quot;,&quot;A&quot;))
ix = which.min(res$padj) # most significant
res &lt;- res[order(res$padj),] # sort
kable(res[1:5,-(3:4)])
</code></pre>

<table>
<thead>
<tr>
<th align="left"></th>
<th align="right">baseMean</th>
<th align="right">log2FoldChange</th>
<th align="right">pvalue</th>
<th align="right">padj</th>
</tr>
</thead>

<tbody>
<tr>
<td align="left">gene9348</td>
<td align="right">48.01943</td>
<td align="right">1.915941</td>
<td align="right">0.0000155</td>
<td align="right">0.1546441</td>
</tr>

<tr>
<td align="left">gene928</td>
<td align="right">60.03093</td>
<td align="right">-1.562622</td>
<td align="right">0.0001151</td>
<td align="right">0.5748158</td>
</tr>

<tr>
<td align="left">gene931</td>
<td align="right">439.23135</td>
<td align="right">-1.101161</td>
<td align="right">0.0004187</td>
<td align="right">0.6066182</td>
</tr>

<tr>
<td align="left">gene4483</td>
<td align="right">16.83153</td>
<td align="right">2.957899</td>
<td align="right">0.0005487</td>
<td align="right">0.6066182</td>
</tr>

<tr>
<td align="left">gene4612</td>
<td align="right">29.09139</td>
<td align="right">-1.742999</td>
<td align="right">0.0006893</td>
<td align="right">0.6066182</td>
</tr>
</tbody>
</table>

<pre><code class="language-r">barplot(assay(dds)[ix,],las=2, main=rownames(dds)[ ix  ]  )
</code></pre>

<p><img src="/post/2018-07-02-deseq2-experimental-design-and-interpretation_files/figure-html/unnamed-chunk-24-1.png" width="576" /></p>

<h2 id="the-condition-effect-for-genotype-iii">The condition effect for genotype III.</h2>

<p>This is the main effect <em>plus</em> the interaction term
(the extra condition effect in genotype III compared to genotype I).</p>

<pre><code class="language-r">res = results(dds, contrast=list( c(&quot;condition_B_vs_A&quot;,&quot;genotypeIII.conditionB&quot;) ))
ix = which.min(res$padj) # most significant
res &lt;- res[order(res$padj),] # sort
kable(res[1:5,-(3:4)])
</code></pre>

<table>
<thead>
<tr>
<th align="left"></th>
<th align="right">baseMean</th>
<th align="right">log2FoldChange</th>
<th align="right">pvalue</th>
<th align="right">padj</th>
</tr>
</thead>

<tbody>
<tr>
<td align="left">gene316</td>
<td align="right">63.666427</td>
<td align="right">-1.724235</td>
<td align="right">0.0000294</td>
<td align="right">0.2936262</td>
</tr>

<tr>
<td align="left">gene252</td>
<td align="right">5.751827</td>
<td align="right">-6.012712</td>
<td align="right">0.0001418</td>
<td align="right">0.2977597</td>
</tr>

<tr>
<td align="left">gene2068</td>
<td align="right">16.851863</td>
<td align="right">-2.725616</td>
<td align="right">0.0001422</td>
<td align="right">0.2977597</td>
</tr>

<tr>
<td align="left">gene6828</td>
<td align="right">19.669827</td>
<td align="right">2.840392</td>
<td align="right">0.0000635</td>
<td align="right">0.2977597</td>
</tr>

<tr>
<td align="left">gene7420</td>
<td align="right">85.226424</td>
<td align="right">-1.510240</td>
<td align="right">0.0001788</td>
<td align="right">0.2977597</td>
</tr>
</tbody>
</table>

<pre><code class="language-r">barplot(assay(dds)[ix,],las=2, main=rownames(dds)[ ix  ]  )
</code></pre>

<p><img src="/post/2018-07-02-deseq2-experimental-design-and-interpretation_files/figure-html/unnamed-chunk-25-1.png" width="576" /></p>

<h2 id="the-condition-effect-for-genotype-ii">The condition effect for genotype II.</h2>

<p>This is the main effect <em>plus</em> the interaction term
(the extra condition effect in genotype II compared to genotype I).</p>

<pre><code class="language-r">res = results(dds, contrast=list( c(&quot;condition_B_vs_A&quot;,&quot;genotypeII.conditionB&quot;) ))
ix = which.min(res$padj) # most significant
res &lt;- res[order(res$padj),] # sort
kable(res[1:5,-(3:4)])
</code></pre>

<table>
<thead>
<tr>
<th align="left"></th>
<th align="right">baseMean</th>
<th align="right">log2FoldChange</th>
<th align="right">pvalue</th>
<th align="right">padj</th>
</tr>
</thead>

<tbody>
<tr>
<td align="left">gene980</td>
<td align="right">71.929692</td>
<td align="right">-1.711667</td>
<td align="right">0.0005822</td>
<td align="right">0.648323</td>
</tr>

<tr>
<td align="left">gene1177</td>
<td align="right">109.057633</td>
<td align="right">-1.281177</td>
<td align="right">0.0002111</td>
<td align="right">0.648323</td>
</tr>

<tr>
<td align="left">gene5118</td>
<td align="right">3.487054</td>
<td align="right">-5.757413</td>
<td align="right">0.0005110</td>
<td align="right">0.648323</td>
</tr>

<tr>
<td align="left">gene5570</td>
<td align="right">23.786589</td>
<td align="right">2.329791</td>
<td align="right">0.0004286</td>
<td align="right">0.648323</td>
</tr>

<tr>
<td align="left">gene6519</td>
<td align="right">94.137270</td>
<td align="right">1.422193</td>
<td align="right">0.0001399</td>
<td align="right">0.648323</td>
</tr>
</tbody>
</table>

<pre><code class="language-r">barplot(assay(dds)[ix,],las=2, main=rownames(dds)[ ix  ]  )
</code></pre>

<p><img src="/post/2018-07-02-deseq2-experimental-design-and-interpretation_files/figure-html/unnamed-chunk-26-1.png" width="576" /></p>

<p>This is equivalent to using numeric vector referencing to each of the results.</p>

<p>[1] &ldquo;Intercept&rdquo;              &ldquo;genotype_II_vs_I&rdquo;       &ldquo;genotype_III_vs_I&rdquo;<br />
[4] &ldquo;condition_B_vs_A&rdquo;       &ldquo;genotypeII.conditionB&rdquo;  &ldquo;genotypeIII.conditionB&rdquo;</p>

<pre><code class="language-r">res = results(dds, contrast= c(0,0,0,1,1,0))
ix = which.min(res$padj) # most significant
res &lt;- res[order(res$padj),] # sort
kable(res[1:5,-(3:4)])
</code></pre>

<table>
<thead>
<tr>
<th align="left"></th>
<th align="right">baseMean</th>
<th align="right">log2FoldChange</th>
<th align="right">pvalue</th>
<th align="right">padj</th>
</tr>
</thead>

<tbody>
<tr>
<td align="left">gene980</td>
<td align="right">71.929692</td>
<td align="right">-1.711667</td>
<td align="right">0.0005822</td>
<td align="right">0.648323</td>
</tr>

<tr>
<td align="left">gene1177</td>
<td align="right">109.057633</td>
<td align="right">-1.281177</td>
<td align="right">0.0002111</td>
<td align="right">0.648323</td>
</tr>

<tr>
<td align="left">gene5118</td>
<td align="right">3.487054</td>
<td align="right">-5.757413</td>
<td align="right">0.0005110</td>
<td align="right">0.648323</td>
</tr>

<tr>
<td align="left">gene5570</td>
<td align="right">23.786589</td>
<td align="right">2.329791</td>
<td align="right">0.0004286</td>
<td align="right">0.648323</td>
</tr>

<tr>
<td align="left">gene6519</td>
<td align="right">94.137270</td>
<td align="right">1.422193</td>
<td align="right">0.0001399</td>
<td align="right">0.648323</td>
</tr>
</tbody>
</table>

<pre><code class="language-r">barplot(assay(dds)[ix,],las=2, main=rownames(dds)[ ix  ]  )
</code></pre>

<p><img src="/post/2018-07-02-deseq2-experimental-design-and-interpretation_files/figure-html/unnamed-chunk-27-1.png" width="576" /></p>

<h2 id="the-effect-of-iii-vs-ii-under-condition-a">The effect of III vs. II, under condition A.</h2>

<p>This is the  effect of &ldquo;genotype_III_vs_I&rdquo;  <em>minus</em>     &ldquo;genotype_II_vs_I&rdquo; ?</p>

<pre><code class="language-r">res = results(dds, contrast= c(0,-1,1,0,0,0))
ix = which.min(res$padj) # most significant
res &lt;- res[order(res$padj),] # sort
kable(res[1:5,-(3:4)])
</code></pre>

<table>
<thead>
<tr>
<th align="left"></th>
<th align="right">baseMean</th>
<th align="right">log2FoldChange</th>
<th align="right">pvalue</th>
<th align="right">padj</th>
</tr>
</thead>

<tbody>
<tr>
<td align="left">gene7875</td>
<td align="right">172.84101</td>
<td align="right">1.316870</td>
<td align="right">0.0000354</td>
<td align="right">0.3541800</td>
</tr>

<tr>
<td align="left">gene316</td>
<td align="right">63.66643</td>
<td align="right">1.399247</td>
<td align="right">0.0006044</td>
<td align="right">0.5256726</td>
</tr>

<tr>
<td align="left">gene557</td>
<td align="right">17.92683</td>
<td align="right">-1.980097</td>
<td align="right">0.0008418</td>
<td align="right">0.5256726</td>
</tr>

<tr>
<td align="left">gene1243</td>
<td align="right">51.92628</td>
<td align="right">-1.844387</td>
<td align="right">0.0004523</td>
<td align="right">0.5256726</td>
</tr>

<tr>
<td align="left">gene1702</td>
<td align="right">55.34857</td>
<td align="right">-1.631481</td>
<td align="right">0.0006698</td>
<td align="right">0.5256726</td>
</tr>
</tbody>
</table>

<pre><code class="language-r">barplot(assay(dds)[ix,],las=2, main=rownames(dds)[ ix  ]  )
</code></pre>

<p><img src="/post/2018-07-02-deseq2-experimental-design-and-interpretation_files/figure-html/unnamed-chunk-28-1.png" width="576" /></p>

<h2 id="the-interaction-term-for-condition-effect-in-genotype-iii-vs-genotype-i">The interaction term for condition effect in genotype III vs genotype I.</h2>

<p>This tests if the condition effect is different in III compared to I</p>

<pre><code class="language-r">res = results(dds, name=&quot;genotypeIII.conditionB&quot;)
ix = which.min(res$padj) # most significant
res &lt;- res[order(res$padj),] # sort
kable(res[1:5,-(3:4)])
</code></pre>

<table>
<thead>
<tr>
<th align="left"></th>
<th align="right">baseMean</th>
<th align="right">log2FoldChange</th>
<th align="right">pvalue</th>
<th align="right">padj</th>
</tr>
</thead>

<tbody>
<tr>
<td align="left">gene9348</td>
<td align="right">48.019429</td>
<td align="right">-2.714891</td>
<td align="right">0.0000104</td>
<td align="right">0.1042074</td>
</tr>

<tr>
<td align="left">gene6251</td>
<td align="right">14.477069</td>
<td align="right">-4.780646</td>
<td align="right">0.0000479</td>
<td align="right">0.2395245</td>
</tr>

<tr>
<td align="left">gene839</td>
<td align="right">161.393935</td>
<td align="right">-1.662756</td>
<td align="right">0.0000886</td>
<td align="right">0.2950638</td>
</tr>

<tr>
<td align="left">gene9383</td>
<td align="right">9.149222</td>
<td align="right">-6.436304</td>
<td align="right">0.0001560</td>
<td align="right">0.3896367</td>
</tr>

<tr>
<td align="left">gene931</td>
<td align="right">439.231351</td>
<td align="right">1.562204</td>
<td align="right">0.0003858</td>
<td align="right">0.4506833</td>
</tr>
</tbody>
</table>

<pre><code class="language-r">barplot(assay(dds)[ix,],las=2, main=rownames(dds)[ ix  ]  )
</code></pre>

<p><img src="/post/2018-07-02-deseq2-experimental-design-and-interpretation_files/figure-html/unnamed-chunk-29-1.png" width="576" /></p>

<h2 id="the-interaction-term-for-condition-effect-in-genotype-iii-vs-genotype-ii">The interaction term for condition effect in genotype III vs genotype II.</h2>

<p>This tests if the condition effect is different in III compared to II</p>

<pre><code class="language-r">res = results(dds, contrast=list(&quot;genotypeIII.conditionB&quot;, &quot;genotypeII.conditionB&quot;))
ix = which.min(res$padj) # most significant
res &lt;- res[order(res$padj),] # sort
kable(res[1:5,-(3:4)])
</code></pre>

<table>
<thead>
<tr>
<th align="left"></th>
<th align="right">baseMean</th>
<th align="right">log2FoldChange</th>
<th align="right">pvalue</th>
<th align="right">padj</th>
</tr>
</thead>

<tbody>
<tr>
<td align="left">gene5761</td>
<td align="right">27.797507</td>
<td align="right">4.231182</td>
<td align="right">0.0000012</td>
<td align="right">0.0116510</td>
</tr>

<tr>
<td align="left">gene6598</td>
<td align="right">100.143591</td>
<td align="right">2.317486</td>
<td align="right">0.0000214</td>
<td align="right">0.1068006</td>
</tr>

<tr>
<td align="left">gene3491</td>
<td align="right">325.224707</td>
<td align="right">1.942243</td>
<td align="right">0.0001168</td>
<td align="right">0.2853814</td>
</tr>

<tr>
<td align="left">gene4635</td>
<td align="right">8.411707</td>
<td align="right">5.206116</td>
<td align="right">0.0000963</td>
<td align="right">0.2853814</td>
</tr>

<tr>
<td align="left">gene5118</td>
<td align="right">3.487054</td>
<td align="right">8.088922</td>
<td align="right">0.0002454</td>
<td align="right">0.2853814</td>
</tr>
</tbody>
</table>

<pre><code class="language-r">barplot(assay(dds)[ix,],las=2, main=rownames(dds)[ ix  ]  )
</code></pre>

<p><img src="/post/2018-07-02-deseq2-experimental-design-and-interpretation_files/figure-html/unnamed-chunk-30-1.png" width="576" /></p>

<h1 id="example-5-three-conditions-three-genotypes-with-interaction-terms">Example 5: Three conditions, three genotypes with interaction terms</h1>

<p>For a more complex example, we now have a 3x3 factorial design. It is convenent to use a diagram to represent study design and the coefficients generated by DESeq2. Each connections in the diagram below represents a coefficient for a contrast and we can use the combination of these to define desired contrasts by travelling through the graph/network.
<img src="I:/factors5.jpg" alt="" /></p>

<pre><code class="language-r">dds &lt;- makeExampleDESeqDataSet(n=10000,m=18)
dds$genotype &lt;- factor(rep(rep(c(&quot;I&quot;,&quot;II&quot;,&quot;III&quot;),each=3),2))
dds$condition &lt;-factor(rep(c(&quot;A&quot;,&quot;B&quot;,&quot;C&quot;), 6)  )
colnames(dds) &lt;- paste(as.character( dds$genotype), as.character( dds$condition),rownames(colData(dds)), sep=&quot;_&quot;  )
colnames(dds) = gsub(&quot;sample&quot;,&quot;&quot;,colnames(dds))
design(dds) &lt;- ~ genotype + condition + genotype:condition
dds &lt;- DESeq(dds)
kable( colData(dds))
</code></pre>

<table>
<thead>
<tr>
<th align="left"></th>
<th align="left">condition</th>
<th align="left">genotype</th>
<th align="right">sizeFactor</th>
</tr>
</thead>

<tbody>
<tr>
<td align="left">I_A_1</td>
<td align="left">A</td>
<td align="left">I</td>
<td align="right">1.055853</td>
</tr>

<tr>
<td align="left">I_B_2</td>
<td align="left">B</td>
<td align="left">I</td>
<td align="right">1.037228</td>
</tr>

<tr>
<td align="left">I_C_3</td>
<td align="left">C</td>
<td align="left">I</td>
<td align="right">1.048146</td>
</tr>

<tr>
<td align="left">II_A_4</td>
<td align="left">A</td>
<td align="left">II</td>
<td align="right">1.048901</td>
</tr>

<tr>
<td align="left">II_B_5</td>
<td align="left">B</td>
<td align="left">II</td>
<td align="right">1.062858</td>
</tr>

<tr>
<td align="left">II_C_6</td>
<td align="left">C</td>
<td align="left">II</td>
<td align="right">1.036711</td>
</tr>

<tr>
<td align="left">III_A_7</td>
<td align="left">A</td>
<td align="left">III</td>
<td align="right">1.052625</td>
</tr>

<tr>
<td align="left">III_B_8</td>
<td align="left">B</td>
<td align="left">III</td>
<td align="right">1.033361</td>
</tr>

<tr>
<td align="left">III_C_9</td>
<td align="left">C</td>
<td align="left">III</td>
<td align="right">1.040028</td>
</tr>

<tr>
<td align="left">I_A_10</td>
<td align="left">A</td>
<td align="left">I</td>
<td align="right">1.054525</td>
</tr>

<tr>
<td align="left">I_B_11</td>
<td align="left">B</td>
<td align="left">I</td>
<td align="right">1.045331</td>
</tr>

<tr>
<td align="left">I_C_12</td>
<td align="left">C</td>
<td align="left">I</td>
<td align="right">1.036150</td>
</tr>

<tr>
<td align="left">II_A_13</td>
<td align="left">A</td>
<td align="left">II</td>
<td align="right">1.052502</td>
</tr>

<tr>
<td align="left">II_B_14</td>
<td align="left">B</td>
<td align="left">II</td>
<td align="right">1.047103</td>
</tr>

<tr>
<td align="left">II_C_15</td>
<td align="left">C</td>
<td align="left">II</td>
<td align="right">1.063802</td>
</tr>

<tr>
<td align="left">III_A_16</td>
<td align="left">A</td>
<td align="left">III</td>
<td align="right">1.053762</td>
</tr>

<tr>
<td align="left">III_B_17</td>
<td align="left">B</td>
<td align="left">III</td>
<td align="right">1.041255</td>
</tr>

<tr>
<td align="left">III_C_18</td>
<td align="left">C</td>
<td align="left">III</td>
<td align="right">1.048660</td>
</tr>
</tbody>
</table>

<pre><code class="language-r">resultsNames(dds)
</code></pre>

<pre><code>## [1] &quot;Intercept&quot;              &quot;genotype_II_vs_I&quot;      
## [3] &quot;genotype_III_vs_I&quot;      &quot;condition_B_vs_A&quot;      
## [5] &quot;condition_C_vs_A&quot;       &quot;genotypeII.conditionB&quot; 
## [7] &quot;genotypeIII.conditionB&quot; &quot;genotypeII.conditionC&quot; 
## [9] &quot;genotypeIII.conditionC&quot;
</code></pre>

<p>To derive desired contrasts, we can travel on the network of factor levels, using the given 9 coefficients as bridges.For example, we are interested in the difference between genotpes III vs. II, under condition C. This is not readily available. But we can travel from C to III, then III to I, I to II, and then II to C. Namely the path is C&ndash;III&ndash;I&ndash;II&ndash;C. In terms of coefficients: (III-I) - (II-I) + III.C - II.C=
      &ldquo;genotype_III_vs_I&rdquo; - &ldquo;genotype_II_vs_I&rdquo;  + &ldquo;genotypeIII.conditionC&rdquo; - &ldquo;genotypeII.conditionC&rdquo;
For such a complex contrast, we can use the numerical combination using a vector:
c(0,-1,1,0,0,0,0,-1,1 )
These numbers are applied to the coefficients in the resultsNames. To double check we order them:</p>

<pre><code class="language-r">cbind(resultsNames(dds),c(0,-1,1,0,0,0,0,-1,1 ) )
</code></pre>

<pre><code>##       [,1]                     [,2]
##  [1,] &quot;Intercept&quot;              &quot;0&quot; 
##  [2,] &quot;genotype_II_vs_I&quot;       &quot;-1&quot;
##  [3,] &quot;genotype_III_vs_I&quot;      &quot;1&quot; 
##  [4,] &quot;condition_B_vs_A&quot;       &quot;0&quot; 
##  [5,] &quot;condition_C_vs_A&quot;       &quot;0&quot; 
##  [6,] &quot;genotypeII.conditionB&quot;  &quot;0&quot; 
##  [7,] &quot;genotypeIII.conditionB&quot; &quot;0&quot; 
##  [8,] &quot;genotypeII.conditionC&quot;  &quot;-1&quot;
##  [9,] &quot;genotypeIII.conditionC&quot; &quot;1&quot;
</code></pre>

<p>Now we can extract the contrast.</p>

<h2 id="the-effect-of-iii-vs-ii-under-condition-c">The effect of III vs. II, under condition C.</h2>

<pre><code class="language-r">res = results(dds, contrast= c(0,-1,1,0,0,0,0,0,1 ))
ix = which.min(res$padj) # most significant
res &lt;- res[order(res$padj),] # sort
kable(res[1:5,-(3:4)])
</code></pre>

<table>
<thead>
<tr>
<th align="left"></th>
<th align="right">baseMean</th>
<th align="right">log2FoldChange</th>
<th align="right">pvalue</th>
<th align="right">padj</th>
</tr>
</thead>

<tbody>
<tr>
<td align="left">gene793</td>
<td align="right">50.368566</td>
<td align="right">3.737810</td>
<td align="right">3.36e-05</td>
<td align="right">0.1460573</td>
</tr>

<tr>
<td align="left">gene1161</td>
<td align="right">827.243573</td>
<td align="right">1.844200</td>
<td align="right">4.39e-05</td>
<td align="right">0.1460573</td>
</tr>

<tr>
<td align="left">gene3314</td>
<td align="right">23.430794</td>
<td align="right">4.580687</td>
<td align="right">1.57e-05</td>
<td align="right">0.1460573</td>
</tr>

<tr>
<td align="left">gene4445</td>
<td align="right">51.054925</td>
<td align="right">-3.096770</td>
<td align="right">6.96e-05</td>
<td align="right">0.1736784</td>
</tr>

<tr>
<td align="left">gene1555</td>
<td align="right">5.073803</td>
<td align="right">10.771498</td>
<td align="right">9.70e-05</td>
<td align="right">0.1937244</td>
</tr>
</tbody>
</table>

<pre><code class="language-r">barplot(assay(dds)[ix,],las=2, main=rownames(dds)[ ix  ]  )
</code></pre>

<p><img src="/post/2018-07-02-deseq2-experimental-design-and-interpretation_files/figure-html/unnamed-chunk-33-1.png" width="576" /></p>

<p>How about many factors, each with many levels? It&rsquo;s going to become increasingly complex. Drawing a network graph like the one above can help you find the right combination.</p>

<p>Please let me know if you have any comment.</p>

    </div>

    


<div class="article-tags">
  
  <a class="btn btn-primary btn-outline" href="/tags/rna-seq/">RNA-Seq</a>
  
</div>




    
    

    

    
<section id="comments">
  <div id="disqus_thread"></div>
<script>
    var disqus_config = function () {
    
    
    
    };
    (function() {
        if (["localhost", "127.0.0.1"].indexOf(window.location.hostname) != -1) {
            document.getElementById('disqus_thread').innerHTML = 'Disqus comments not available by default when the website is previewed locally.';
            return;
        }
        var d = document, s = d.createElement('script'); s.async = true;
        s.src = '//' + "https-gex-netlify-com" + '.disqus.com/embed.js';
        s.setAttribute('data-timestamp', +new Date());
        (d.head || d.body).appendChild(s);
    })();
</script>
<noscript>Please enable JavaScript to view the <a href="https://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
<a href="https://disqus.com" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>
</section>



  </div>
</article>

<footer class="site-footer">
  <div class="container">
    <p class="powered-by">

      &copy; 2018 &middot; 

      Powered by the
      <a href="https://sourcethemes.com/academic/" target="_blank" rel="noopener">Academic theme</a> for
      <a href="https://gohugo.io" target="_blank" rel="noopener">Hugo</a>.

      <span class="pull-right" aria-hidden="true">
        <a href="#" id="back_to_top">
          <span class="button_icon">
            <i class="fa fa-chevron-up fa-2x"></i>
          </span>
        </a>
      </span>

    </p>
  </div>
</footer>


<div id="modal" class="modal fade" role="dialog">
  <div class="modal-dialog">
    <div class="modal-content">
      <div class="modal-header">
        <button type="button" class="close btn-large" data-dismiss="modal">&times;</button>
        <h4 class="modal-title">Cite</h4>
      </div>
      <div>
        <pre><code class="modal-body tex"></code></pre>
      </div>
      <div class="modal-footer">
        <a class="btn btn-primary btn-outline js-copy-cite" href="#" target="_blank">
          <i class="fa fa-copy"></i> Copy
        </a>
        <a class="btn btn-primary btn-outline js-download-cite" href="#" target="_blank">
          <i class="fa fa-download"></i> Download
        </a>
        <div id="modal-error"></div>
      </div>
    </div>
  </div>
</div>

    

    
    
    <script id="dsq-count-scr" src="//https-gex-netlify-com.disqus.com/count.js" async></script>
    

    

    
    <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.2.1/jquery.min.js" integrity="sha512-3P8rXCuGJdNZOnUx/03c1jOTnMn3rP63nBip5gOP2qmUh5YAdVAvFZ1E+QLZZbC1rtMrQb+mah3AfYW11RUrWA==" crossorigin="anonymous"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery.imagesloaded/4.1.3/imagesloaded.pkgd.min.js" integrity="sha512-umsR78NN0D23AzgoZ11K7raBD+R6hqKojyBZs1w8WvYlsI+QuKRGBx3LFCwhatzBunCjDuJpDHwxD13sLMbpRA==" crossorigin="anonymous"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha512-iztkobsvnjKfAtTNdHkGVjAYTrrtlC7mGp/54c40wowO7LhURYl3gVzzcEqGl/qKXQltJ2HwMrdLcNUdo+N/RQ==" crossorigin="anonymous"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery.isotope/3.0.4/isotope.pkgd.min.js" integrity="sha512-VDBOIlDbuC4VWxGJNmuFRQ0Li0SKkDpmGyuhAG5LTDLd/dJ/S0WMVxriR2Y+CyPL5gzjpN4f/6iqWVBJlht0tQ==" crossorigin="anonymous"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/fancybox/3.2.5/jquery.fancybox.min.js" integrity="sha256-X5PoE3KU5l+JcX+w09p/wHl9AzK333C4hJ2I9S5mD4M=" crossorigin="anonymous"></script>
    
    
    <script src="/js/hugo-academic.js"></script>
    

    
    
      
      
      <script src="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/highlight.min.js" integrity="sha256-/BfiIkHlHoVihZdc6TFuj7MmJ0TWcWsMXkeDFwhi0zw=" crossorigin="anonymous"></script>
      

      

      

      <script>hljs.initHighlightingOnLoad();</script>
    

    
    

  </body>
</html>

